### PR TITLE
feature: allow course instructors to remove pupils from course

### PR DESCRIPTION
**Pull request by dhenkel92**
image
image

### CODE REVIEW

**Comment by dhenkel92** [comment]
https://github.com/corona-school/backend/pull/860

**Comment by dhenkel92** [comment]
https://github.com/corona-school/project-user/issues/947

File: src/pages/student/SingleCourseStudent.tsx [hunk 1]
```diff
@@ -262,6 +286,20 @@ const SingleCourseStudent = () => {
...
+					mutation removeParticipantFromCourse($subcourseId: Float!, $pupilId: Float!) {
+						subcourseLeave(subcourseId: $subcourseId, pupilId: $pupilId)
+					}
+				`)
+    );
+
+    const doRemoveParticipant = async (pupilId: number) => {
+        await removeParticipant({ variables: { subcourseId: subcourseId, pupilId: pupilId } });
+        toast.show({ description: t('single.removeParticipantFromCourseModal.success'), placement: 'top' });
+        refetchBasics();
```
Comment (Jonasdoubleyou) [inline src/pages/student/SingleCourseStudent.tsx:300]: Are the participants part of the basics query? Don't we also need to refetch the instructor only query?

**Review by Jonasdoubleyou — DISMISSED** [review]

File: src/pages/student/SingleCourseStudent.tsx (hunk 1 above)
Comment (dhenkel92) [reply src/pages/student/SingleCourseStudent.tsx:300]: The participant list was updated with this command. Not sure if we need the instructor query as well.
I'll have a look.

Comment (Jonasdoubleyou) [reply src/pages/student/SingleCourseStudent.tsx:300]: A maybe refetchBasics transiently causes the other query to also refetch. If it works it's fine :)

Comment (dhenkel92) [reply src/pages/student/SingleCourseStudent.tsx:300]: I think i'll change it to the instructor query as we are fetching the participants there. I'm not feeling too comfortable relying on the side-effect of refetchBasics 😅

**Review by Jonasdoubleyou — APPROVED** [review]

**Pull request merged**